### ISSUE TITLE
Problem: no project style guide

### ISSUE THREAD

**Issue by gazhayes** [Blockrazor/blockrazor#934]
Problem: there's no style guide for working on blockrazor

solution: have a look at existing js, es6, node and meteor style guides, piece together something that works well with the C4.

**Comment by AshHimself** [comment]
You talking about a possible linter ?

**Comment by gazhayes** [comment]
No, linters etc hide code smells.

I mean an actual style guide, here's an example: http://zeromq.org/docs:style

**Comment by AshHimself** [comment]
Right! This should add another layer of fun for new contributors but totally understand the need :D

**Comment by gazhayes** [comment]
The C4 says you *should* follow the style guide if there is one, so a patch is still valid if it doesn't follow it.

**Comment by AshHimself** [comment]
The style guide should state to remove debugging console.log statements before PR, so I don't have to remove useless messages spamming my console when I am trying to debug :D

**Comment by anbud** [comment]
I agree with this one, I fully understand the need. :D 
It would be lovely to achieve at least some consistency throughout the code.

**Comment by gazhayes** [comment]
@janat08 this is a reference from another project for what I mean: http://zeromq.org/docs:style

**Comment by janat08** [comment]
Beautifyrc lib tries to format code after parameters in .beautifyrc file, I think I snuck in one of those into upstream at some point, and meteor guide touches on code style.

**Comment by gazhayes** [comment]
We shouldn't do any code reformatting, it hides code smells.

**Comment by janat08** [comment]
Do you mean using 2 spaces for indendation and actually using indentation, because that what I keep referring to- I read the whole issue. https://github.com/marketplace/styleci

**Comment by anbud** [comment]
Just to add, it's not a bad idea to use a CI reformatter with a limited set of rules (indentation, whitespaces, stuff like that), just for aesthetics and consistency.
If we're going to use a CI reformatter, there's no need to pay the monthly fee for StyleCI, we can set up our own thing with eslint or js-beautify, and existing Travis CI infrastructure.

**Comment by janat08** [comment]
run on lines in html are common that are then fairly inconsistent.

**Comment by gazhayes** [comment]
@janat08 if you want to set up a basic reformatter as @anbud  suggests that's ok.

**Comment by janat08** [comment]
I'm not interested.

**Comment by anbud** [comment]
Since @janat08 is not interested, should I take it?

**Comment by gazhayes** [comment]
Sure, if you want to. It's probably not very high priority but indentation variance is starting to get pretty annoying.

**Comment by anbud** [comment]
I've taken some time to research it and there's only way to make this work, Travis would have to push beautifier's changes directly to this repo, which, I believe, breaks C4 as it would involve writing directly to master.
@gazhayes, what are your thoughts on this one?

**Comment by gazhayes** [comment]
Yeah it would break the C4. OK, another option is for contributors to just run it themselves every now and then on (problem: weird indentation everywhere. solution: run beautifier).

**Comment by anbud** [comment]
You're right. I'll just implement a simple js-beautify script then that users can run. :)

**Comment by anbud** [comment]
Done in #1034. We should probably choose a custom eslint style config, as choosing the standard style says we have about 30000 lint errors of which about 4000 can't be automatically fixed.
Running the linter is done with  `npm run lint`.

**Comment by AshHimself** [comment]
So we should be running npm run lint now? I tried it and got a million errors :D

**Comment by janat08** [comment]
https://github.com/beautify-web/js-beautify
just reformats code.
